src/tests/suite/daos_bigio.c

```diff
@@ -0,0 +1,505 @@
+	iod.iod_eprs	= NULL;
+	iod.iod_csums	= NULL;
+	iod.iod_type	= DAOS_IOD_ARRAY;
+
+	/** update record */
+	print_message("writing %lu records of %lu bytes each at offset %lu\n",
+		      recx.rx_nr, iod.iod_size, recx.rx_idx);
+	rc = daos_obj_update(oh, DAOS_TX_NONE, &dkey, 1, &iod, &sgl, NULL);
+	print_message("daos_obj_update() returns %d\n", rc);
+	assert_int_equal(rc, 0);
+
+	/** fetch the records back */
+	buf_out = calloc(arg->size, arg->nr);
+	assert_non_null(buf_out);
+
+	d_iov_set(&sg_iov, buf_out, arg->size * arg->nr);
+	sgl.sg_iovs = &sg_iov;
+
+	iod.iod_size = DAOS_REC_ANY;
+
+	print_message("reading data back\n");
+	rc = daos_obj_fetch(oh, DAOS_TX_NONE, &dkey, 1, &iod, &sgl, NULL, NULL);
+	print_message("daos_obj_fetch() returns %d\n", rc);
+	assert_int_equal(rc, 0);
+
+	/** verify record size */
+	print_message("validating record size ...\n");
+	assert_int_equal(iod.iod_size, arg->size);
+
+	/** Verify data consistency */
+	print_message("validating data ...\n");
+	assert_memory_equal(buf, buf_out, arg->size * arg->nr);
+
+	/** close object */
+	rc = daos_obj_close(oh, NULL);
+	assert_int_equal(rc, 0);
+
+	rc = daos_cont_close(coh, NULL);
+	assert_int_equal(rc, 0);
+
+	while (arg->myrank == 0) {
+		rc = daos_cont_destroy(arg->pool.poh, co_uuid, 1, NULL);
+		if (rc == -DER_BUSY) {
+			print_message("Container is busy, wait\n");
+			sleep(1);
+			continue;
+		}
+		break;
+	}
+
+	if (arg->multi_rank)
+		MPI_Bcast(&rc, 1, MPI_INT, 0, MPI_COMM_WORLD);
+	if (rc) {
+		print_message("failed to destroy container "DF_UUIDF
+			      ": %d\n", DP_UUID(co_uuid), rc);
+		assert_int_equal(rc, 0);
+	}
+
+	free(buf_out);
+	free(buf);
+	print_message("all good\n");
+}
+
+static void
+bigio_noncontig(void **state)
+{
+	test_arg_t	*arg = *state;
+	daos_obj_id_t	 oid;
+	daos_handle_t	 oh;
+	d_iov_t	 dkey;
+	d_sg_list_t	 sgl;
+	d_iov_t	 sg_iov;
+	daos_iod_t	 iod;
+	daos_recx_t	*recxs;
+	int		*buf;
+	int		*buf_out;
+	daos_size_t	 i;
+	uuid_t		 co_uuid;
+	daos_cont_info_t co_info;
+	daos_handle_t	 coh;
+	int		 rc;
+
+	if (arg->myrank == 0) {
+		uuid_generate(co_uuid);
+		print_message("setup: creating container "DF_UUIDF"\n",
+			      DP_UUID(co_uuid));
+		rc = daos_cont_create(arg->pool.poh, co_uuid, NULL, NULL);
+		if (rc) {
+			print_message("daos_cont_create failed, rc: %d\n", rc);
+			goto bcast;
+		}
+
+		rc = daos_cont_open(arg->pool.poh, co_uuid, DAOS_COO_RW,
+				    &coh, &co_info, NULL);
+		if (rc)
+			print_message("daos_cont_open failed, rc: %d\n", rc);
+	}
+
+bcast:
+	if (arg->multi_rank)
+		MPI_Bcast(&rc, 1, MPI_INT, 0, MPI_COMM_WORLD);
+	assert_int_equal(rc, 0);
+
+	if (arg->multi_rank)
+		handle_share(&coh, HANDLE_CO, arg->myrank, arg->pool.poh, 0);
+
+	arg->size = 4;
+	/*
+	 * 50 GB P0ol size
+	 *
+	 * at 30MB:
+	 * CRIT src/vos/vos_tree.c:282 kb_rec_alloc() assertion failure
+	 * kb_rec_alloc: Assertion `ta != ((void *)0)' failed.
+	 *
+	 * > 30MB:
+	 * -1007 - NO SPACE
+	 */
+	arg->nr = (daos_size_t)(1024*1024) * 50 / arg->size;
```

Comment:
again should check the status of this and update the comments.
also make sure we have the correct pool size created.

Reply:
Will verify. Maybe my test machine has low tmpfs storage space.